### PR TITLE
Passing args to docker run CLI

### PR DESCRIPTION
**Pull request by ksanjeevan**
## What changes are proposed in this pull request?
cc @tomasatdatabricks 

From the discussion in https://github.com/mlflow/mlflow/issues/2607, ~this is more of a draft to see how passing user arguments to `docker run` could be implemented. No tests or documentation has been written for now.~

I've used the docker example to test this works. Simply building the image in that example and then running:
```
mlflow run examples/docker -P alpha=0.5 -A gpus=all
``` 

Shows how by propagating this flag to the `docker run` the container can now talk to the gpu:

```
=============
== PyTorch ==
=============

NVIDIA Release 20.02 (build 10315040)
PyTorch Version 1.5.0a0+3bbb36e

Container image Copyright (c) 2019, NVIDIA CORPORATION.  All rights reserved.

Copyright (c) 2014-2019 Facebook Inc.
Copyright (c) 2011-2014 Idiap Research Institute (Ronan Collobert)
Copyright (c) 2012-2014 Deepmind Technologies    (Koray Kavukcuoglu)
Copyright (c) 2011-2012 NEC Laboratories America (Koray Kavukcuoglu)
Copyright (c) 2011-2013 NYU                      (Clement Farabet)
Copyright (c) 2006-2010 NEC Laboratories America (Ronan Collobert, Leon Bottou, Iain Melvin, Jason Weston)
Copyright (c) 2006      Idiap Research Institute (Samy Bengio)
Copyright (c) 2001-2004 Idiap Research Institute (Ronan Collobert, Samy Bengio, Johnny Mariethoz)
Copyright (c) 2015      Google Inc.
Copyright (c) 2015      Yangqing Jia
Copyright (c) 2013-2016 The Caffe contributors
All rights reserved.

Various files include modifications (c) NVIDIA CORPORATION.  All rights reserved.
NVIDIA modifications are covered by the license terms that apply to the underlying project or file.

NOTE: MOFED driver for multi-node communication was not detected.
      Multi-node communication performance may be reduced.

NOTE: The SHMEM allocation limit is set to the default of 64MB.  This may be
   insufficient for PyTorch.  NVIDIA recommends the use of the following flags:
   nvidia-docker run --ipc=host ...

Elasticnet model (alpha=0.500000, l1_ratio=0.100000):
  RMSE: 0.7947931019036528
  MAE: 0.6189130834228137
  R2: 0.1841166871822183
Cuda available: True
```

### Is this a user-facing change?

- [ ] UI
- [x] CLI
- [ ] API
- [ ] REST-API
- [ ] Examples
- [ ] Docs
- [ ] Tracking
- [x] Projects
- [ ] Artifacts
- [ ] Models
- [ ] Scoring
- [ ] Serving
- [ ] R
- [ ] Java
- [x] Python

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [x] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Review by tomasatdatabricks — COMMENTED** [review]
Hi @ksanjeevan. This looks good. 

Can you add tests? I think it would be great to get this in.

**Comment by ksanjeevan** [comment]
Awesome @tomasatdatabricks, I'll get on that and keep updating the pr.

**Comment by ksanjeevan** [comment]
Hey @tomasatdatabricks, let me know what you think. I wrote a small test checking that the `docker_args` were converted correctly in the docker command, would you like to see anything else tested? Also I'm not doing any checking for the user to pass valid flags.

I have a small example that could go into the examples as `examples/docker_cuda` or something like that if that's of interest, lmk.

File: mlflow/cli.py [hunk 1]
```diff
@@ -112,6 +115,20 @@ def run(uri, entry_point, version, param_list, experiment_name, experiment_id, b
             eprint("Repeated parameter: '%s'" % name)
             sys.exit(1)
         param_dict[name] = value
+
+    args_dict = {}
+    for s in docker_args:
```
Comment (tomasatdatabricks) [inline mlflow/cli.py]: We should probably extract this into a shared function.

**Review by tomasatdatabricks — COMMENTED** [review]
I left a small comment, otherwise looks good!

**Comment by ksanjeevan** [comment]
@tomasatdatabricks Addressed! Let me know if there's anything else.

**Review by tomasatdatabricks — APPROVED** [review]
lgtm! thanks @ksanjeevan

**Comment by ksanjeevan** [comment]
great!

**Comment by CPFelix** [comment]
@ksanjeevan When I use "mlflow.run" instead of "mlflow run",how can I do?

**Pull request merged**